File: NERP.NetworkEntities/Models/Duct.cs
using NERP.Common.Models;

namespace NERP.NetworkEntities.Models
{
    public class Duct : BaseNetworkEntity
    {
        public LineString Geometry { get; set; }
        public double? MeasuredLength { get; set; }
        public double? CalculatedLength { get; set; }
        public double? Height { get; set; }
        public double? Width { get; set; }
        public double? InnerDiameter { get; set; }
        public double? OuterDiameter { get; set; }
        public Installation Installation { get; set; }
        public Owner Owner { get; set; }
    }
}

File: NERP.Admin.Web/ClientApp/src/app/entities/models/duct.ts
import { LineString } from '../../common/geo/geometry';
import { BaseNetworkEntity } from './baseNetworkEntity';
import { Installation } from './installation';
import { Owner } from './owner';

export class Duct extends BaseNetworkEntity {

    height?: number;
    width?: number;
    innerDiameter?: number;
    outerDiameter?: number;

    geometry: LineString;
    measuredLength?: number;
    calculatedLength?: number;

    installation?: Installation = new Installation();
    owner?: Owner = new Owner();
}

